src/ucp/core/ucp_am.c

```diff
@@ -203,14 +203,15 @@ ucs_status_t ucp_worker_set_am_recv_handler(ucp_worker_h worker,
         goto out;
     }
 
-    worker->am.cbs[id].cb         = param->cb; /* cb should always be set (can be NULL) */
-    worker->am.cbs[id].context    = (param->field_mask &
-                                     UCP_AM_HANDLER_ATTR_FIELD_ARG) ?
-                                    param->arg : NULL;
-    worker->am.cbs[id].flags      = (param->field_mask &
-                                     UCP_AM_HANDLER_ATTR_FIELD_FLAGS) ?
-                                    param->flags : 0;
-    worker->am.cbs[id].priv_flags = UCP_AM_CB_PRIV_FLAG_NBX;
+    /* cb should always be set (can be NULL) */
+    ucs_array_elem(&worker->am, id).cb         = param->cb;
+    ucs_array_elem(&worker->am, id).context    = UCP_PARAM_VALUE(AM_HANDLER,
+                                                                 param, arg,
+                                                                 ARG, NULL);
+    ucs_array_elem(&worker->am, id).flags      = UCP_PARAM_VALUE(AM_HANDLER,
+                                                                 param, flags,
+                                                                 FLAGS, 0);
+    ucs_array_elem(&worker->am, id).priv_flags = UCP_AM_CB_PRIV_FLAG_NBX;
```

Comment:
maybe use local var: `am_cb = &ucs_array_elem(&worker->am, id)`
(in several places we use multiple entries from the callback handler sruct)